### PR TITLE
mitosheet: make log anon opt in vs opt out

### PR DESCRIPTION
**Pull request by naterush**
# Description

This PR makes it so you have to be explicit about what data we're logging. This will be slightly annoying when introducing new steps, but it will have the benefit of making it _very clear_ to both us and our users that we aren't taking any private data. This, I think, is great!

The other thing this does is actually raise exceptions if our logging fails. Like our APIs, for some reason we decided to hold our logging to a lower standard than other pieces of the codebase. I don't like when errors fail silently!

# Testing

The goal here is to use the front-end to touch every step. I tried to exhaustively do this, but try as well. I also think we should leave this in dev for the entirety of notebook testing, so we can make sure no more issues appear in the wild. 

Also, _go and look at all the logs you generate_. This takes a while (at least an hour or so), but generating these logs and then looking at them is the only way we can make sure that we're actually capturing the information we want, and no more!

# Documentation

Nope.

### CODE REVIEW

File: mitosheet/mitosheet/api/api.py [hunk 1]
```diff
@@ -113,28 +113,29 @@ def handle_api_event(
     so that the frontend knows how to match the responses.
     """
     result: Union[str, List[str]]
+    params = event['params']
     if event["type"] == "datafiles":
-        result = get_datafiles(event)
+        result = get_datafiles(params)
```
Comment (naterush) [inline mitosheet/mitosheet/api/api.py:118]: TODO: go through each of these functions and rename event -> params

File: mitosheet/mitosheet/telemetry/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Telemetry in Mitosheet
+
+We don't log any of your private data, but we do collect telemetry information about button clicks, where Mito is loaded from, and other general interactions with the app.
```
Comment (naterush) [inline mitosheet/mitosheet/telemetry/README.md:3]: TODO: commit the new stuff?

File: mitosheet/mitosheet/telemetry/telemetry_utils.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
+import platform
```
Comment (naterush) [inline mitosheet/mitosheet/telemetry/telemetry_utils.py:1]: TODO: go through file line by line

File: mitosheet/mitosheet/telemetry/private_params_map.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+
+# Params that do not need to be anonyimized
```
Comment (naterush) [inline mitosheet/mitosheet/telemetry/private_params_map.py:7]: TODO: document file

File: mitosheet/mitosheet/telemetry/anonymization_utils.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+from mitosheet.user.schemas import UJ_USER_SALT
+
+
+valid_words = ['cat', 'dog', 'hat', 'time', 'person', 'year', 'way', 'thing', 'man', 'world', 'life', 'born', 'part', 'child', 'eye', 'woman', 'place', 'work', 'fall', 'case', 'point', 'company', 'number', 'group', 'problem', 'fact']
+
+# We use the same salt to anonymize_words, and we read
+# this salt in once the function is called for the first
+# time, to make sure it's initialized properly
+salt = None
+def anonymize_word(word: Any) -> str:
```
Comment (naterush) [inline mitosheet/mitosheet/telemetry/anonymization_utils.py:19]: TODO: go through file line by line

**Review by naterush — COMMENTED** [review]
Cleanup!

File: mitosheet/mitosheet/telemetry/telemetry_utils.py [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+    # it is given to many values).
+    if start_time is not None:
+        processing_time = time.perf_counter() - start_time
+        processing_time_params['processing_time'] = round(processing_time, 1)
+        processing_time_params['processing_time_seconds'] = int(round(processing_time, 0))
+        processing_time_params['processing_time_seconds_ten'] = int(round(processing_time, -1))
+        processing_time_params['processing_time_seconds_hundred'] = int(round(processing_time, -2))
+
+        # If we just did an update, and this update has a pandas processing time, then we can calculate the
+        # time that we spent as mito overhead vs. just
```
Comment (aarondr77) [inline mitosheet/mitosheet/telemetry/telemetry_utils.py:154]: This sentence is unfinished

**Comment by aarondr77** [comment]
I got a sheet crashing error when I renamed a graph

Screen Shot 2022-04-30 at 11 09 37 AM

Screen Shot 2022-04-30 at 11 10 04 AM

**Comment by aarondr77** [comment]
On a rename, `params_new_dataframe_name` displays the actual name

**Comment by aarondr77** [comment]
The filter location param is sometimes empty for both the filter/sort tab and the values tab

Screen Shot 2022-04-30 at 11 28 15 AM

**Comment by aarondr77** [comment]
Besides for that, everything looks good and I don't see any personal data coming through. Thanks for prioritizing our user's safety

File: mitosheet/mitosheet/telemetry/anonymization_utils.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    if key in LOG_PARAMS_PUBLIC:
+        private_params[key] = value
+    elif key in LOG_PARAMS_TO_ANONYIMIZE:
+        private_params[key] = anonyimize_object(value)
+    elif key in LOG_PARAMS_FORMULAS:
+        private_params[key] = anonymize_formula(value, params['sheet_index'], steps_manager)
+    else:
+        raise Exception('key, value not in any log set', key, value)
+    
+    return private_params
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/mitosheet/telemetry/anonymization_utils.py:115]: So smart, so great. I love it.

**Pull request merged**